### PR TITLE
7 interactive plot with leaflet

### PR DESCRIPTION
**Pull request by BaptisteDlp**
Keypoints for this PR:

- Add dynamic plot mode in both `plotStorms()` and plot `plotBehaviour()`

### CODE REVIEW

File: R/plotStorms.R [hunk 1]
```diff
@@ -2,6 +2,42 @@
 
 
 
+#' Get type for icon (leaflet)
+#'
+#' @noRd
+#' @param msw numeric
+#'
+#' @return corresponding icon
+getType <- function(msw) {
```
Comment (thomasarsouze) [inline R/plotStorms.R:11]: I think this function is not used. Needs to be removed

Comment (BaptisteDlp) [reply R/plotStorms.R]: Done, I forgot indeed

File: data-raw/internalData.R [hunk 2]
```diff
@@ -34,6 +34,15 @@ groundColor <- "grey"
 
 sshsPalette <- c("#00CCFF", "#00CCCC", "#FFFFB2", "#FECC5C", "#FD8D3C", "#F03B20", "#BD0026")
 
+SSHS_PALETTE <- c("#00CCFF", "#00CCCC", "#FFFFB2", "#FECC5C", "#FD8D3C", "#F03B20", "#BD0026")
```
Comment (thomasarsouze) [inline data-raw/internalData.R:37]: Why do you need this one, it is already defined just above ?
```suggestion
```

File: data-raw/internalData.R [hunk 3]
```diff
@@ -34,6 +34,15 @@ groundColor <- "grey"
...
+names(SSHS_PALETTE) <- c("Tropical Depression",
```
Comment (thomasarsouze) [inline data-raw/internalData.R:38]:
```suggestion
names(sshsPalette) <- c("Tropical Depression",
```

File: data-raw/internalData.R [hunk 4]
```diff
@@ -62,6 +71,7 @@ usethis::use_data(resolutions,
     km, wgs84, Basins, sshs,
     margin,
     oceanColor, groundColor, sshsPalette, mswSSHSPalette, mswPalette, pdiPalette, exposurePalette,
+    SSHS_PALETTE,
```
Comment (thomasarsouze) [inline data-raw/internalData.R:74]:
```suggestion
```

File: R/plotStorms.R [hunk 5]
```diff
@@ -288,54 +298,138 @@ plotStorms <- function(sts,
...
+      )
+    }
+    
+    
+    #Adding legends
+    map <- leaflet::addLegend(map,
+                              legends,
+                              colors = SSHS_PALETTE,
+                              labels = names(SSHS_PALETTE),
+                              title = "SSHS",
```
Comment (thomasarsouze) [inline R/plotStorms.R:426]:
```suggestion
                              colors = sshsPalette,
                              labels = names(sshsPalette),
                              title = "SSHS",
```

File: data-raw/internalData.R (hunk 2 above)
Comment (BaptisteDlp) [reply data-raw/internalData.R:37]: I would like to rename all constants in capital letters but you told me that you prefer keeping them in lower case

Comment (BaptisteDlp) [reply data-raw/internalData.R:37]: Done

File: data-raw/internalData.R (hunk 4 above)
Comment (BaptisteDlp) [reply data-raw/internalData.R]: Done

File: tests/testthat/test-plotStorms.R [hunk 6]
```diff
@@ -2,452 +2,597 @@
...
+      legends = "topleft",
+      loi = TRUE,
+      xlim = NULL,
+      ylim = NULL,
+      dynamicPlot = c(TRUE, TRUE)
+    )
+  )
 })
+
+# TODO do not know why it is not working on remote repo
```
Comment (thomasarsouze) [inline tests/testthat/test-plotStorms.R:591]: Actually not working on my computer localy:
```
expect_equal(map,mapPam)
Erreur : `map` (`actual`) not equal to `mapPam` (`expected`).

attr(actual$x$calls[[2]]$args[[1]], 'bbox') vs attr(expected$x$calls[[2]]$args[[1]], 'bbox')
                                                          [,1]      [,2]
- attr(actual$x$calls[[2]]$args[[1]], 'bbox')[1, ]   163.65362 172.82097
+ attr(expected$x$calls[[2]]$args[[1]], 'bbox')[1, ] 163.70071 172.82097
- attr(actual$x$calls[[2]]$args[[1]], 'bbox')[2, ]   -22.98754 -10.31482
+ attr(expected$x$calls[[2]]$args[[1]], 'bbox')[2, ] -22.99300 -10.32286

  `attr(actual$x$calls[[2]]$args[[1]][[1]][[1]][[1]], 'row.names')[682:684]`: 682 683 684                             and 11 more...
`attr(expected$x$calls[[2]]$args[[1]][[1]][[1]][[1]], 'row.names')[682:702]`: 682 683 684 685 686 687 688 689 6
```

File: R/plotStorms.R [hunk 7]
```diff
@@ -288,54 +298,142 @@ plotStorms <- function(sts,
...
+
+    #Adding legends
+    map <- leaflet::addLegend(map,
+                              legends,
+                              colors = sshsPalette,
+                              labels = names(sshsPalette),
+                              title = "SSHS",
+                              opacity = 1)
+
+    map
```
Comment (thomasarsouze) [inline R/plotStorms.R:433]: Why do you need this line ?

**Comment by thomasarsouze** [comment]
Please also update vignettes with examples of dynamical plots, and precising zooming / clicking options

**Comment by BaptisteDlp** [comment]
> Please also update vignettes with examples of dynamical plots, and precising zooming / clicking options

Done, Thomas could you please do the cross references between two sections in SpatialBehaviour vignette ?

File: R/plotStorms.R (hunk 7 above)
Comment (BaptisteDlp) [reply R/plotStorms.R:433]: I don't remember for the moment, I think it is actually what creates the plot

**Pull request merged**